spec/display_spec.rb

```diff
@@ -0,0 +1,56 @@
+describe Display do
+  describe '#view_menu' do
+    it 'shows the menu in nice format' do
+      expect(subject.view_menu(Menu::PIZZA)).to include("Marinara: £8.00")
+    end
+  end
+
+  describe '#view_order' do
+    let(:order) { [{ bufalina: 13 }, { diavola: 14 }] }
+
+    it 'displays the order in nice format' do
+      expect(subject.view_order(order)).to include("Diavola: £14.00")
+    end
+
+    it 'displays the total in nice format' do
+      expect(subject.view_order(order)).to include("Total: £27.00")
+    end
+  end
+
+  describe '#confirm_added' do
+    it 'confirms meal was added to order' do
+      confirm_add = described_class::MEAL_ADD_CONFIRMATION
+      expect(subject.confirm_added('marinara')).to eq "Marinara#{confirm_add}"
+    end
+  end
+
+  describe '#confirm_order' do
+    context 'when customer cancels' do
+      it 'throws error' do
+        allow(STDOUT).to receive(:puts) { nil }
+        allow(subject).to receive(:gets) { 'c' }
+        expect { subject.confirm_order }.to raise_error RuntimeError
+      end
+    end
+
+    context 'when customer confirms' do
+      let(:confirmation) do
+        [described_class::CONFIRMATION_PROMPT,
+         described_class::ORDER_CONFIRMATION].join("\n") + "\n"
+      end
+
+      it 'confirms the order' do
+        allow(STDOUT).to receive(:puts) { nil }
+        allow(subject).to receive(:gets) { "\n" }
+        expect { subject.confirm_order }.to output(confirmation).to_stdout
+      end
+    end
+  end
+
+  describe '#invalid_meal' do
+    it 'tells customer choice is invalid' do
+      invalid = "chips#{described_class::INVALID_MEAL_MESSAGE}"
```

Comment:
Rather than mimicking what the body of the method looks like, in tests we want to be as explicit and human readable as possible for what the expected output should be.

For example, let's say I wrote a method `sum` which takes two values as input and returns the sum of them. In my tests, I shouldn't have something that looks like:
```
def add(x,y)
    x + y
end
describe "#sum"
    it "adds two numbers together correctly" do
        @total = 3 + 4
        expect(sum(3,4)).to eq @total
    end
end
```

Because you're just recreating the logic of the code which risks the test passing even if there is a bug in the code as this bug will exist in both the original method and the expected output of the test. For example, the following would pass because I've mimicked the bug in my test, even though my logic is incorrect.
```
# actual code
def sum(x,y)
    x - y
end
# tests
describe "#sum"
    it "adds two numbers together correctly" do
        @total = 3 - 4
        expect(sum(3,4)).to eq @total
    end
end
```
To avoid this, the test for `sum` should look like, this way I am not mimicking the code in the test and I will catch any bugs.
```
describe "#sum"
    it "adds two numbers together correctly" do
        expect(sum(3,4)).to eq 7
    end
end
```

Reply:
so instead of using the variable name `"#{described_class::INVALID_MEAL_MESSAGE}"`
it's preferable to hardcode the desired output?

Reply:
Yes, hardcording it would make it more readable. It would also mean that the test would fail if anyone changed the constant's value which I think is a nice side effect of this change, as it will make people more conscious of changing the constant's value